concent_api/core/transfer_operations.py

```diff
@@ -277,3 +277,22 @@ def send_request_to_storage_cluster(headers, request_http_address, method='head'
         headers=headers,
         stream=stream,
     )
+
+
+def calculate_token_expiration_deadline(
+    operation: FileTransferToken.Operation,
+    report_computed_task: message.tasks.ReportComputedTask,
+) -> int:
+    if operation == FileTransferToken.Operation.upload:
+        token_expiration_deadline = (
+            report_computed_task.task_to_compute.compute_task_def['deadline'] +
+            3 * settings.CONCENT_MESSAGING_TIME +
+            2 * calculate_maximum_download_time(report_computed_task.size)
```

Comment:
I have discovered this today while testing verification, function `calculate_maximum_download_time` requires also `rate` parameter.

Reply:
@rwrzesien `Rate`? Where can I locate this parameter? Because i search in CTD, TTC and RTC and none of this messages have this.

Reply:
Done